data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/AvroConsumer.java

```diff
@@ -83,6 +90,30 @@ public void consumeRecords(final KafkaConsumer<String, GenericRecord> consumer,
         }
     }
 
+    private void iterateRecordPartitions(KafkaSourceBufferAccumulator kafkaSourceBufferAccumulator,
+                                         ConsumerRecords<String, GenericRecord> records) throws Exception {
+        for (TopicPartition partition : records.partitions()) {
+            List<Record<Object>> kafkaRecords = new ArrayList<>();
+            List<ConsumerRecord<String, GenericRecord>> partitionRecords = records.records(partition);
+            iterateConsumerRecords(kafkaSourceBufferAccumulator, kafkaRecords, partitionRecords);
+            if (!kafkaRecords.isEmpty()) {
+                kafkaSourceBufferAccumulator.writeAllRecordToBuffer(kafkaRecords, buffer, topicConfig);
+            }
+        }
+    }
+
+    private void iterateConsumerRecords(KafkaSourceBufferAccumulator kafkaSourceBufferAccumulator,
+                                        List<Record<Object>> kafkaRecords,
+                                        List<ConsumerRecord<String, GenericRecord>> partitionRecords) {
+        for (ConsumerRecord<String, GenericRecord> consumerRecord : partitionRecords) {
+            lastReadOffset = kafkaSourceBufferAccumulator.processConsumerRecords(offsetsToCommit, kafkaRecords, lastReadOffset, consumerRecord, partitionRecords);
```

Comment:
Earlier there was toString() to get json from GenericRecord. is it not required now?